### DISCUSSION TITLE
Are "React Compiler" and "initialize state with props" compatible?

### DISCUSSION THREAD

**Discussion by yslpn** [pmndrs/zustand#2807] [Q&A]
Hi. I wanted to ask how the store initialization pattern with props and the new React Compiler fit together? Is there any way to avoid this error in the linter and how safe is it to turn off the linter in this case?

https://zustand.docs.pmnd.rs/guides/initialize-state-with-props

I often use this pattern in my code:

```javascript
export function MySuperProvider({ myProps, children }: MySuperProps) {
  const storeRef = useRef(createHostStore(myProps));

  return (
    <MySuperContext.Provider value={storeRef.current}>
      {children}
    </MySuperContext.Provider>
  );
}
```

But I get an eslint error:

```
Ref values (the `current` property) may not be accessed during render. (https://react.dev/reference/react/useRef)eslint(react-compiler/react-compiler)
```

**Comment by dbritto-dev** [comment]
@yslpn some docs are outdated, I encourage you to read our new docs at `hooks`, `apis`, and `middlewares`

**Reply by dbritto-dev** [reply]
@yslpn btw, you could to this:

```ts
const [store] = useState(() => createHostStore(myProps))
```

**Comment by dai-shi** [comment]
https://x.com/dai_shi/status/1846480778689135014

It's still uncertain if the error is a true positive or a false positive.
